### PR TITLE
Update jenkins-core to 2.414.3 (#678)

### LINKED ISSUE #678: Update jenkins-core to 2.414.3
**Describe the feature request**
As a maintainer I want to keep my dependencies up to date.

**Acceptance criteria**
- [x] update core dependency to 2.414.3
- [x] update all other dependencies
  - [x] only dependencies where the api did not change significantly, 
  - [x] otherwise the latest version without significant changes
- [x] update allowlist
- [x] ensure compat with latest LTS

### PR DESCRIPTION
**Pull request by MarioFuchsTT**
Resolves #678

### CODE REVIEW

File: config/allowed_licenses.json [hunk 1]
```diff
@@ -10,10 +10,10 @@
     },
     {
       "moduleName": "org.xerial:sqlite-jdbc",
-      "moduleVersion": "3.7.2",
+      "moduleVersion": "3.42.0.1",
       "moduleLicense": "Apache-2.0",
       "moduleCopyright": "David Crawshaw, 2006",
-      "licenseUrl": "https://github.com/xerial/sqlite-jdbc/blob/sqlite-jdbc-3.7.2/LICENSE.zentus",
+      "licenseUrl": "https://github.com/xerial/sqlite-jdbc/blob/3.42.0.1/LICENSE.zentus",
```
Comment (ErikRehmTT) [inline config/allowed_licenses.json:16]: This file does not contain the Apache 2.0 license 

Apache 2.0 Licence is located at `https://github.com/xerial/sqlite-jdbc/blob/3.42.0.1/LICENSE`

File: config/allowed_licenses.json [hunk 2]
```diff
@@ -242,10 +234,10 @@
     },
     {
       "moduleName": "commons-codec:commons-codec",
-      "moduleVersion": "1.15",
+      "moduleVersion": "1.16.0",
       "moduleLicense": "Apache-2.0",
-      "moduleCopyright": "2002-2020, The Apache Software Foundation",
-      "licenseUrl": "https://github.com/apache/commons-codec/blob/rel/commons-codec-1.15/NOTICE.txt",
+      "moduleCopyright": "2002-2023 The Apache Software Foundation",
+      "licenseUrl": "https://github.com/apache/commons-codec/blob/rel/commons-codec-1.16.0/NOTICE.txt",
```
Comment (ErikRehmTT) [inline config/allowed_licenses.json:240]: https://github.com/apache/commons-codec/blob/rel/commons-codec-1.16.0/LICENSE.txt

File: config/allowed_licenses.json (hunk 1 above)
Comment (MarioFuchsTT) [reply config/allowed_licenses.json]: resolved in [9aeb2c1](https://github.com/jenkinsci/ecutest-plugin/pull/690/commits/9aeb2c12c224ff250cc72a24b8015930fb3ba03c)

File: config/allowed_licenses.json (hunk 2 above)
Comment (MarioFuchsTT) [reply config/allowed_licenses.json]: resolved in [9aeb2c1](https://github.com/jenkinsci/ecutest-plugin/pull/690/commits/9aeb2c12c224ff250cc72a24b8015930fb3ba03c)

**Review by ErikRehmTT — APPROVED** [review]

**Pull request merged**